### ISSUE TITLE
BUG: "Added recipe folder cleanup and protection" somehow prevents twitch from updating or deleting any modpack containing this mod

### ISSUE THREAD

**Issue by SirAron111** [SleepyTrousers/EnderIO#5268]
#### Issue Description:
“Added recipe folder cleanup and protection” somehow prevents twitch from updating or deleting any modpack uses this mod newest version. (maybe its something else you did but this seemes to be the most fitting thing how did you even protect it and from what)

So first I thought it’s an issue with twitch. That if you try to delete a profile or update an older one it says files are locked or accessed elsewhere you can’t delete it/ update it now.
But this issue comes from the ender io config recipe folder. How the fuck did he figure that out you ask easy. I am a patient guy and I tested it multiple times once I found it out to make sure it’s really this.

My way of doing it was to make a copy of my modpack profile. Than make sure that deleting it via the twitch option doesn’t work for that copy as well (couldn’t delete the copy as well). To narrow the error down first I tried to delete all mods. This didn’t fix it. Then I deleted all folders one by one. And every folder I deleted I tried updating or deleting the profile. Until I deleted the configs folder and it finally worked. So I made a new copy and this time I one by one deleted all the mod configs first all the .cfg files. This didn’t help coldnt delete the copy. Then I went to the folders deleting them fixed it again so I made another copy of my profile than started deleting all the config folders of the mods inside the config folder one by one until deleting the ender io one did it. And again new profile copy starts deleting the files in the ender io config folder one by one ….. I repeated this trial and error method until I finally found it. Deleting the recipe folder inside the ender io folder fixes it. So I tried deleting the files inside that one, one by one again but only after it was completely empty I could delete the modpack profile. (of course deleting the Modpack instance folder manually always worked but it seems that if you try it via the twitch apps normal updating button or delete button it thinks the folder is locked or protected and can’t do shit)….

If you need reproduction steps try it yourself. Your best bet is installing my modpack as it has the new version and this issue. But I think you will need to make a world and do something related to ender io first be it creative mode or not for it to trigger this bug.

#### What happens:
Like stated above. New version of ender io when played with for a bit you can’t delete or update modpacks containing it anymore as the recipe folder for whatever reason prevents twitch to access/overwrite/delete it in any way so you need to manually kill it from the PC.

#### What you expected to happen:
Being able to simply Update my modpack over twitch like I always do as well as having all my players be able to update without having to make new profiles  where they need to copy pasta all their save files into and then still being unable to delete the old one until they manually delete the Folder from the PC.

#### Steps to reproduce:
1. Play a modpack with your mods newest version in it (or an older one and add the version yourself), for a bit do some enderio stuff maybe (via the twitch app) try my pack Dimension Zero version 9.5 it has the issue version in it and later you can try updating it to 9.6 version after loading it up for a bit first which popped up the issue of being unable to update or delete it since players noticed it there first.
2. Try deleting the profile (or a copy so you don’t have to play again if it doesn’t trigger not sure what triggers it i just know the game needs to read from the folder once)
3. If you suddenly can’t delete the profile via the app itself try deleting the recipe folder in ender io and bam can be deleted or updated again.
.........Strangest bug I ever had to narrow down on.

____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: EnderIO-1.12.2-5.1.52.jar
- EnderCore: EnderCore-1.12.2-0.5.72.jar
- Minecraft: 1.12.2
- Forge: 14.23.5.2847 
- SpongeForge? yes/no ---
- Optifine? yes/no ---
- Single Player and/or Server? ---

#### Your most recent log file where the issue was present: 
Not really a log file as it’s not a crash or a bug in game it just seems like whatever you did with the config folder doesn’t work right, or keeps access to it blocked even for things that it shouldn’t block like the Modpack updating/deleting feature from twitch. (It’s only half an issue with twitch as it’s only your mod causing this problem but I did report the issue to them as well)

PS: if you need more information just ask.

GZ
SirAron

**Comment by HenryLoenwind** [comment]
You need to report that to twitch. Their launcher's deletion code is wrong. You can tell them to call me if they need someone to explain simple file operations to them.

(Yes, I know what's happening there just from reading the title.)

**Comment by SirAron111** [comment]
O ok I already made a ticket to them but only figured the problem out after that in a (pretty long testing session).

What I think will happen thought is that once more people update to your new version and if twitch doesn’t fix it you will get more and more reports about this (if others find this issue that is took me pretty long to notice where its coming from)

I will make another issue ticket to them linking to this problem here but maybe you should make one as well. Simply because they don’t like me since I report so many issue to them. (I am good at finding bugs after my player’s report them to me and I already had like 6 bugs that came from twitch since I made my pack they had to fix them all XD) (And like over 80 bugs/crashes with mods where I flooded problem pages annoying people like you …SRY)

**Comment by jelaw21** [comment]
![image](https://user-images.githubusercontent.com/21100827/68059000-49523700-fcd1-11e9-8c92-d3bac2b2be75.png)

Response from Twitch when I reported the same issue.

**Comment by SirAron111** [comment]
I hope this is going to be fixable. Updating Modpacks properly is hard like this. Other launchers than twitch seem to have similar issues with the recipes folder. (Makes updating really painful for inexperienced players/end-users).

**Comment by Wxrlds** [comment]
I think what's happening here is EnderIO setting the permissions for the folder to read only, making Twitch unable to update modpacks.

**Comment by SirAron111** [comment]
@Wxrlds Could be but. But as it is it’s just a big annoyance as many people use the twitch ap and not many people understand the error I even made an exclaimer into my update note telling people how to solve it by deleting the ender io config folder and then pressing update, as the updates ender io folder will be added anyway. But for most common players even that is a bit too hard to understand. I hope they make an exception for the app or since the twitch devs don’t seem to care much fixing things themselves calling it unsupported changes, and I can kind of know what they mean I mean imagine every mod maker would make individual protections for files of some sort they would have to keep updating stuff. On the other hand I can understand the Ender IO devs as well they made something for some reason and don’t want to bother changing it back so it fits in line with common stuff like screw the system.

**Comment by Seth0067** [comment]
EnderIO coders are, undoubtedly, far, FAR better than Twitch App team.  However, the platform owners are able to get away with such calls and they make the rules and setv their development priorities.  To put it another way, who fixes the bug - is it Windows or third party app developers? Usually the latter if they want to play within Windows.  

I'm not saying that's what the EnderIO team should do, but no one's expectations should be that Twitch will go out of its way to support a competitor's ecosystem. I would hope the EnderIO team would understand this and mitigate the issue as they can on their end despite the shittiness of the situation, as they are ultimately helping their loyal, core users (packdevs) by doing so.

**Comment by SirAron111** [comment]
@Seth0067 Ye and not only the pack devs I mean they can easily work around this issue by deleting the files themselves its more about all the players of said packs who wish to use the new Ender IO versions but every time the pack updates hey will mostly not be able to update as not everyone who simply plays miencraft after watching their favorite you tuber or so understands how to solve such issues. So for me it’s more about how to help players I mean aren’t mods made so people can play them? Whtas the point if they can’t play them! And it’s not only in twitch I got the same issue when I tried to update via other providers like MultiMC or AT launcher I don’t think every one of them will rework their code that worked fine until now.
